lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/persistence/table/ColumnModelUtilsTest.java

```diff
@@ -363,6 +364,88 @@ public void testNormalizedDoubleEnum() {
+	}
+	
+	@Test
+	public void testNormalizedCloneWithSubColumnsWithValidSubs() {
+		ColumnModel column = new ColumnModel().setName("jsonRoot").setColumnType(ColumnType.JSON)
+				.setJsonSubColumns(List.of(new JsonSubColumnModel().setName("a").setJsonPath("$.a").setColumnType(ColumnType.INTEGER)));
+
+		// call under test
+		ColumnModel normalized = ColumnModelUtils.createNormalizedClone(column,
+				StackConfigurationSingleton.singleton().getTableMaxEnumValues());
+		assertEquals(column.getJsonSubColumns(), normalized.getJsonSubColumns());
+	}
+	
+	@Test
+	public void testNormalizedCloneWithSubColumnsWithInvalidSub() {
+		ColumnModel column = new ColumnModel().setName("jsonRoot").setColumnType(ColumnType.JSON)
+				.setJsonSubColumns(List.of(new JsonSubColumnModel().setName(null).setJsonPath("$.a").setColumnType(ColumnType.INTEGER)));
+
+		String message = assertThrows(IllegalArgumentException.class, ()->{
+			// call under test
+			ColumnModelUtils.createNormalizedClone(column,
+					StackConfigurationSingleton.singleton().getTableMaxEnumValues());
+		}).getMessage();
+		assertEquals("JsonSubColumnModel.name is required.", message);
+	}
+	
+	@Test
+	public void testValidateJsonSubColumn() {
+		JsonSubColumnModel sub = new JsonSubColumnModel().setName("a").setJsonPath("$.a").setColumnType(ColumnType.INTEGER);
+		// call under test
+		ColumnModelUtils.validateJsonSubColumn(sub);
+	}
+	
+	@Test
+	public void testValidateJsonSubColumnWithNull() {
+		JsonSubColumnModel sub = null;
+		String message = assertThrows(IllegalArgumentException.class, ()->{
+			// call under test
+			ColumnModelUtils.validateJsonSubColumn(sub);
+		}).getMessage();
+		assertEquals("JsonSubColumnModel is required.", message);
+	}
+	
+	@Test
+	public void testValidateJsonSubColumnWithNullName() {
+		JsonSubColumnModel sub = new JsonSubColumnModel().setName(null).setJsonPath("$.a").setColumnType(ColumnType.INTEGER);
+		String message = assertThrows(IllegalArgumentException.class, ()->{
+			// call under test
+			ColumnModelUtils.validateJsonSubColumn(sub);
+		}).getMessage();
+		assertEquals("JsonSubColumnModel.name is required.", message);
+	}
+	
+	@Test
+	public void testValidateJsonSubColumnWithNullPath() {
+		JsonSubColumnModel sub = new JsonSubColumnModel().setName("a").setJsonPath(null).setColumnType(ColumnType.INTEGER);
+		String message = assertThrows(IllegalArgumentException.class, ()->{
+			// call under test
+			ColumnModelUtils.validateJsonSubColumn(sub);
+		}).getMessage();
+		assertEquals("JsonSubColumnModel.jsonPath is required.", message);
+	}
+	
+	@Test
+	public void testValidateJsonSubColumnWithNullType() {
+		JsonSubColumnModel sub = new JsonSubColumnModel().setName("a").setJsonPath("$.a").setColumnType(null);
+		String message = assertThrows(IllegalArgumentException.class, ()->{
+			// call under test
+			ColumnModelUtils.validateJsonSubColumn(sub);
+		}).getMessage();
+		assertEquals("JsonSubColumnModel.columnType is required.", message);
+	}
```

Comment:
- Maybe a test where the column type is JSON and there is a facet type (should throw?)?
- Also throw if you have a subcolumn and tyupe is not json
- Also throw if the subcolumn is JSON itself?